controllers/setupcontroller/setup_controller.go

```diff
@@ -0,0 +1,77 @@
+package setupcontroller
+
+import (
+	"context"
+	"fmt"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhClient "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/client"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/upgrade"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	finalizerName = "datasciencecluster.opendatahub.io/finalizer"
+)
+
+type SetupControllerReconciler struct {
+	*odhClient.Client
+}
+
+func (r *SetupControllerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logf.FromContext(ctx).WithName("DataScienceCluster")
+	log.Info("Reconciling setup controller", "Request.Name", req.Name)
+	currentOperatorRelease := cluster.GetRelease()
+	platform := currentOperatorRelease.Name
+	instances := &dscv1.DataScienceClusterList{}
+
+	// Check if the delete ConfigMap exists
+	if !upgrade.HasDeleteConfigMap(ctx, r.Client) {
+		log.Info("No delete ConfigMap found, skipping reconciliation")
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch the DataScienceCluster instance
```

Comment:
I do not see that

Reply:
Wrong place of the comment, I'll fix that